src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import halfMoon from "../assets/half-moon.png"
+import sheep from "../assets/sheep.png"
+
+const ThemeToggle = () => {
+  const [isDark, setIsDark] = useState(false)
+  useEffect(() => {
+    // Checks initial Theme
+    if (document.documentElement.getAttribute("data-theme") == "dark") {
+      console.log("setting")
+      setIsDark(true)
+      localStorage.setItem("theme", "dark")
+    } else {
+      setIsDark(false)
+      localStorage.setItem("theme", "light")
+    }
+  }, [])
+
+  const handleChangeTheme = useCallback(() => {
+    if (localStorage.getItem("theme") === "dark") {
+      document.documentElement.setAttribute('data-theme', 'light');
+      localStorage.setItem("theme", "light")
+      setIsDark(false)
+    } else {
+      document.documentElement.setAttribute('data-theme', 'dark');
+      localStorage.setItem("theme", "dark")
+      setIsDark(true)
+    }
```

Comment:
I think this can be improved to be more DRY. Both branches do essentially the same.